### PR TITLE
Add user configurable extensions

### PR DESCRIPTION
**Pull request by Kidswiss**
## Summary

* expose a list of extensions that can be configured by the user
* tested extensions:
  * postgis
  * timescaledb
  * ltree 

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: apis/stackgres/v1/sgpostresconfig.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// This CRD isn't autogenerated, as the generator from the CRD somewhat falls apart with this one.
+// As it only has two fields it should not be too much trouble.
```
Comment (glrf) [inline apis/stackgres/v1/sgpostresconfig.go:8]: I think we should be able to make this work, but I also struggle to do so right now.  I think this is fine for now, but maybe we should think about investing some time to un-YOLO-fy this CRD generation in the future.

File: pkg/comp-functions/functions/vshn-postgres-func/extensions.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		return runtime.NewFatalErr(ctx, "not able to get cluster", err)
+	}
+
+	finalExtensions := []stackgresv1.SGClusterSpecPostgresExtensionsItem{}
+
+	for _, ext := range extensionsMap {
+		finalExtensions = append(finalExtensions, ext)
+	}
+
+	cluster.Spec.Postgres.Extensions = &finalExtensions
```
Comment (glrf) [inline pkg/comp-functions/functions/vshn-postgres-func/extensions.go:64]: The order when iterating over a map is not fixed. So this `extensions` array will have a different order every time.  Did you check that we do not introduce unnecessary updates of the cluster spec every reconcile loop?

File: apis/stackgres/v1/sgpostresconfig.go (hunk 1 above)
Comment (Kidswiss) [reply apis/stackgres/v1/sgpostresconfig.go:8]: Yeah we'll have to look into this in the future.

File: pkg/comp-functions/functions/vshn-postgres-func/extensions.go (hunk 2 above)
Comment (Kidswiss) [reply pkg/comp-functions/functions/vshn-postgres-func/extensions.go:64]: Good point, I did not check that. But I think with a sort before we apply it, it should be a non-issue.

**Comment by Kidswiss** [comment]
Woops, just realized that I forgot about some tests 🙈 Will add them.

**Review by zugao — APPROVED** [review]
One thing rest looks good

**Review by glrf — APPROVED** [review]

**Pull request merged**